EmployeeManagementService/EmployeeManagementService.Domain/Services/EmployeeService.cs

```diff
@@ -28,5 +28,51 @@ public async Task<List<Employee>> GetAllEmployees(int page, int offset)
 
             return employees.Select(e => EmployeeMapper.ToCoreEmployee(e)).ToList();        
         }
+
+        public async Task<Employee> GetEmployeeById(long id)
+        {
+            var employee = await _employeeRepository.GetEmployeeById(id);
+
+            if (employee == null)
+            {
+                throw new ArgumentException("Employee does not exist.");
+            }
+
+            return EmployeeMapper.ToCoreEmployee(employee);
+        }
+
+        public async Task IncrementEmployeeFailedLoginAttempt(long id)
+        {
+            var employee = await _employeeRepository.GetEmployeeById(id);
+
+            if (employee == null)
+            {
+                throw new ArgumentException("Employee does not exist.");
+            }
+
+            if (employee.FailedLoginAttempts < 3)
+            {
+                await _employeeRepository.IncrementEmployeeFailedLoginAttempt(employee.Id);
+            }
+
+            if (employee.FailedLoginAttempts == 3)

```

Comment:
There is a logic issue here.

Suppose failed login attempts is two and a user tries again. It goes into your first If statement and updates it to 3.

Then it exists the repository if statement and checks is my failed login attempts == to 3. But what is the failed login attempts really equaled to here? In this context?

I suggest modifying the repository method for this to return an int. Return the failed attempts count after you updated.